### PR TITLE
[Codelabs] Update build tests for codelabs

### PR DESCRIPTION
**Pull request by michdud**
Allows building from branch, tag, or commit. Defaults to build only 104 complete and 111 complete.

### CODE REVIEW

File: scripts/build_codelabs [hunk 1]
```diff
@@ -32,24 +34,126 @@ function is_xcpretty_available() {
...
+      shift
+      ;;
+    -h|--help)
+      echo
+      echo "Builds MDC iOS Codelabs."
+      echo
+      echo "The default branch of MDC iOS that is used for testing is your current branch."
+      echo "Your current branch is: $DEFAULT_BRANCH"
+      echo
+      echo "The default codelabs built are 104 Complete and 111 Complete. You can build all with -a."
```
Comment (ianegordon) [inline scripts/build_codelabs]: Consider:
By default the Swift and Objective-C Google I/O 2018 Codelab 104 and Codelab 111 targets will be built.
You can build all codelabs by specifying --all.

File: scripts/build_codelabs [hunk 2]
```diff
@@ -32,24 +34,126 @@ function is_xcpretty_available() {
...
+  fi
+}
+
+# Checks that the argument is a valid reference.
+#
+# Exits if the reference is not valid.
+function check_reference() {
+  reference=$(git ls-remote https://github.com/material-components/material-components-ios.git $1 | wc -l)
+  if [ $reference -eq 0 ]; then
+    echo "Invalid reference name."
```
Comment (ianegordon) [inline scripts/build_codelabs]: Consider:
"Unable to find $1 on GitHub"

File: scripts/build_codelabs [hunk 3]
```diff
@@ -32,24 +34,126 @@ function is_xcpretty_available() {
...
+      ;;
+    -a|--all)
+      all_codelabs=1
+      shift
+      ;;
+    -h|--help)
+      echo
+      echo "Builds MDC iOS Codelabs."
+      echo
+      echo "The default branch of MDC iOS that is used for testing is your current branch."
```
Comment (ianegordon) [inline scripts/build_codelabs]: Consider:
By default the codelabs are tested against the remote branch tracking your current branch.

File: scripts/build_codelabs [hunk 4]
```diff
@@ -32,24 +34,126 @@ function is_xcpretty_available() {
...
+      echo "Your current branch is: $DEFAULT_BRANCH"
+      echo
+      echo "The default codelabs built are 104 Complete and 111 Complete. You can build all with -a."
+      echo
+      echo "Options:"
+      echo "-v,  --verbose              Verbose logging."
+      echo "-rb, --remote-branch        Remote branch of MDC-iOS with which to test codelabs."
+      echo "-c,  --commit               Commit SHA of MDC-iOS with which to test codelabs."
+      echo "-t,  --tag                  Tag of MDC-iOS with which to test codelabs."
+      echo "-a,  --all                  Build all the codelabs."
```
Comment (ianegordon) [inline scripts/build_codelabs:146]: Super-SUPER-nit: Add -h, --help to this menu

**Review by ianegordon — APPROVED** [review]
Consider adding something along the following to every error message.
echo "$(basename($0)) -h for help"

I tried running this from a non-tracked branch and it said "Invalid reference name." which wasn't super obvious.

This is great!

File: scripts/build_codelabs (hunk 1 above)
Comment (michdud) [reply scripts/build_codelabs]: Right now, the script points to the master branch of the codelabs (I can't see an I/O branch there). Should I refer to the codelabs as the Google I/O 2018 codelabs?

Comment (ianegordon) [reply scripts/build_codelabs]: Good point. 
Could you add tag to the current master called googleio2018 at head? (Outside of this PR)
That way we can continue to point at master, but we'll have a reference if we need it.
I think referring to them as the 'codelabs' is fine in the general case, in this specific case I would urge details.  My thinking is that if something goes wrong, the person will end up coming through this path (--help !) and will gain some insight.

**Comment by michdud** [comment]
I'm ready to merge!

**Pull request merged**